src/codemirror/widget.ts

```diff
@@ -102,6 +149,31 @@ class CodeMirrorWidget extends Widget {
     this._editor.focus();
   }
 
+  /**
+   * Handle `focus` events for the widget.
+   */
+  private _evtFocus(event: MouseEvent): void {
+    this._rendered.hide();
+    this._live.show();
+    this._editor.focus();
+  }
+
+  /**
+   * Handle `blur` events for the widget.
+   */
+  private _evtBlur(event: MouseEvent): void {
+    if (this.node.contains(event.relatedTarget as HTMLElement)) {
+      return;
+    }
+    this._live.hide();
+    this._rendered.show();
+    CodeMirror.runMode(this._editor.getDoc().getValue(),
```

Comment:
Should this be done before the node is shown? Or does it require the node to be in the DOM?